retriever.py

```diff
@@ -52,100 +106,80 @@ def __init__(self, experiment_name):
-        #     "bert-base-uncased"
-        # ).to("cuda:0")
-        generator = Llama.build(
-            ckpt_dir=os.path.expanduser(self.config.model_ckpt_dir),
-            tokenizer_path=os.path.expanduser(self.config.tokenizer_path),
-            max_seq_len=self.config.max_context_len,
-            max_batch_size=self.config.max_batch_size,
-            use_kv_cache=False,
-            model_parallel_size=1,
-        )
-        self.model = generator.model
-        self.tokenizer = Tokenizer(self.config)
+    def evaluate_embedding(self, pooled_embeddings):
+        # Takes the pooled embedding and evaluates it against the embeddings
+        # from the indexer
+        mapping = {}
+        for embedding in pooled_embeddings:
+            similarity_matrix = F.cosine_similarity(self.embeddings, embedding)
 
-    def retrieve(self):
-        # parse and tokenize input (function from a file)
-        # run model forward on each chunk of the embeddings
-        # cosine similarity per chunk
-        # rank and print most/least match
-        tensor = torch.full(
-            (1, self.config.max_context_len),
-            self.tokenizer.pad_id,
-            dtype=torch.long,
-        )
+            grouped_by_file = defaultdict(list)
 
-        functions = get_functions(
-            "/home/osalpekar/pytorch/torch/distributed/fsdp/fully_sharded_data_parallel.py",
-            indexing=False,
-            # "/home/osalpekar/pytorch/torch/fx/node.py"
-        )
-        for signature in functions:
-            # Just doing a sample function with the __init__ here
-            if "__init__" in signature:
-                function_body = functions[signature]
-                print(signature)
-                tokens = self.tokenizer.encode(function_body)  # .to("cuda:0")
-                tokens = tokens[: self.config.max_context_len]
-                tensor[0, : len(tokens)] = torch.tensor(
-                    tokens, dtype=torch.long
-                )
-
-        tensor = tensor.to(self.device)
-
-        self.model.eval()
-        with torch.no_grad():
-            # full_model_states = self.model(
-            #     tokens, output_hidden_states=True
-            # )
-            # test_embedding = full_model_states.hidden_states[-1].detach()
-
-            _, func_embedding = self.model.forward(
-                tensor, 0, output_last_hidden_state=True
-            )
-            pooled_embedding = torch.sum(func_embedding, dim=1)
-
-            similarity_matrix = F.cosine_similarity(
-                self.embeddings, pooled_embedding
-            )
-            # Largest scores are last!
-            sorted_indices = torch.argsort(similarity_matrix, descending=False)
-
-            print("Top 10 Most Relevant Tests")
-            top_indices = sorted_indices[-10:]
-            for ind in top_indices:
-                test = self.unittest_names[int(ind.item())]
+            for ind in range(similarity_matrix.shape[0]):
```

Comment:
We don't need the sorted indices anymore, but I'm also not quite sure if this logic is correct.